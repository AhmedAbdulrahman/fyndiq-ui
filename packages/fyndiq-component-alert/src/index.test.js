import React from 'react'
import { shallow, mount } from 'enzyme'
import Alert from './'

describe('fyndiq-component-alert', () => {
  test('should be rendered without props', () => {
    expect(shallow(<Alert>Notification</Alert>)).toMatchSnapshot()
  })

  test('should have a type prop', () => {
    expect(shallow(<Alert type="warning">Warning</Alert>)).toMatchSnapshot()
    expect(shallow(<Alert type="good">Good</Alert>)).toMatchSnapshot()
    expect(shallow(<Alert type="bad">Bad</Alert>)).toMatchSnapshot()
  })

  test('should have unclosable prop', () => {
    expect(shallow(<Alert unclosable />)).toMatchSnapshot()
  })

  test('should be self-closable', () => {
    jest.useFakeTimers()
    const component = mount(<Alert />)
    component.find('.close').simulate('click')
    jest.runTimersToTime(50)
    expect(component).toMatchSnapshot()
    jest.runTimersToTime(500)
    expect(component).toMatchSnapshot()
  })

  test('should have a onClose handler', () => {
    const closeSpy = jest.fn()
    const component = mount(<Alert onClose={closeSpy} />)
    component.find('.close').simulate('click')
    expect(closeSpy.mock.calls).toHaveLength(1)
  })

  test('stopShowingAfter and stopShowingAfterKey props functionality', () => {
    const props = {
      stopShowingAfter: 3,
      stopShowingAfterKey: 'Information-Alert',
    }
    const reactComponent = <Alert {...props} />
    let component = mount(reactComponent)
    component.find('.close').simulate('click')
    component = mount(reactComponent)
    component.find('.close').simulate('click')
    component = mount(reactComponent)
    component.find('.close').simulate('click')
    component = mount(reactComponent)
    expect(component).toMatchSnapshot()
  })
})
