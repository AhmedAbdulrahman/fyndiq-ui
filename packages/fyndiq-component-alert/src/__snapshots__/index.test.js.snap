// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fyndiq-component-alert should be rendered without props 1`] = `
<div
  className="
            alertWrapper
            
            
          "
>
  <div
    className="
                alert
                type-info
              "
  >
    <span
      className="text"
    >
      Notification
    </span>
    <div
      className="close"
      onClick={[Function]}
    >
      ×
    </div>
  </div>
</div>
`;

exports[`fyndiq-component-alert should be self-closable 1`] = `
<Alert
  onClose={[Function]}
  stopShowingAfterKey=""
  type="info"
  unclosable={false}
>
  <div
    className="
              alertWrapper
              hidden
              
            "
  >
    <div
      className="
                  alert
                  type-info
                "
    >
      <span
        className="text"
      />
      <div
        className="close"
        onClick={[Function]}
      >
        ×
      </div>
    </div>
  </div>
</Alert>
`;

exports[`fyndiq-component-alert should be self-closable 2`] = `
<Alert
  onClose={[Function]}
  stopShowingAfterKey=""
  type="info"
  unclosable={false}
>
  <div
    className="
              alertWrapper
              hidden
              removed
            "
  >
    <div
      className="
                  alert
                  type-info
                "
    >
      <span
        className="text"
      />
      <div
        className="close"
        onClick={[Function]}
      >
        ×
      </div>
    </div>
  </div>
</Alert>
`;

exports[`fyndiq-component-alert should have a type prop 1`] = `
<div
  className="
            alertWrapper
            
            
          "
>
  <div
    className="
                alert
                type-warning
              "
  >
    <span
      className="text"
    >
      Warning
    </span>
    <div
      className="close"
      onClick={[Function]}
    >
      ×
    </div>
  </div>
</div>
`;

exports[`fyndiq-component-alert should have a type prop 2`] = `
<div
  className="
            alertWrapper
            
            
          "
>
  <div
    className="
                alert
                type-good
              "
  >
    <span
      className="text"
    >
      Good
    </span>
    <div
      className="close"
      onClick={[Function]}
    >
      ×
    </div>
  </div>
</div>
`;

exports[`fyndiq-component-alert should have a type prop 3`] = `
<div
  className="
            alertWrapper
            
            
          "
>
  <div
    className="
                alert
                type-bad
              "
  >
    <span
      className="text"
    >
      Bad
    </span>
    <div
      className="close"
      onClick={[Function]}
    >
      ×
    </div>
  </div>
</div>
`;

exports[`fyndiq-component-alert should have unclosable prop 1`] = `
<div
  className="
            alertWrapper
            
            
          "
>
  <div
    className="
                alert
                type-info
              "
  >
    <span
      className="text"
    />
  </div>
</div>
`;

exports[`fyndiq-component-alert stopShowingAfter and stopShowingAfterKey props functionality 1`] = `
<Alert
  onClose={[Function]}
  stopShowingAfter={3}
  stopShowingAfterKey="Information-Alert"
  type="info"
  unclosable={false}
>
  <div
    className="
              alertWrapper
              
              
            "
  >
    <div
      className="
                  alert
                  type-info
                "
    >
      <span
        className="text"
      />
      <div
        className="close"
        onClick={[Function]}
      >
        ×
      </div>
    </div>
  </div>
</Alert>
`;
